### ISSUE TITLE
AutoUpdater quitAndInstall is quitting the app after downloading the update but update is not happening

### ISSUE THREAD

**Issue by Sai1919** [electron-userland/electron-builder#4891]
electron-builder: "22.4.0"
electron-updater: "4.2.5"
electron: "8.1.1" (current)
target: Windows

This is more of a query than issue at this point,
Earlier I was using the following versions:
"electron-updater": "2.21.10"
"electron": "3.1.0"
"electron-builder": "19.47.1"

I have a "close" handler on the Browser window where I call "event.preventDefault()" and show a pop up for the user to confirm his action of closing window and upon his confirmation I either close or keep the window. The code looks something like this:
```
 window.on('close', (event) => {
    event.preventDefault()
    showConfirmationBox()
  })
```
The autoupdates were working absolutely fine with this code until I upgrade to the latest electron, electron-builder and electron-updater versions that I mentioned at the top. I found that the issue is because of the code in 'close' handler getting triggered when I call "autoUpdater.quitAndInstall(true, true)" so I handled it accordingly and everything is working fine now but I want to know why is it working with older versions and not with the new versions.

Please let me know if you need anymore information.

Thanks,

**Comment by hhyyg** [comment]
Same problem. I solved this way: https://github.com/electron-userland/electron-builder/issues/4143#issuecomment-521850797

**Comment by Sai1919** [comment]
> Same problem. I solved this way: [#4143 (comment)](https://github.com/electron-userland/electron-builder/issues/4143#issuecomment-521850797)

I actually skipped calling "event.preventDefault()" if the app is quitting because of calling "quitAndInstall" so it fixed the problem but wanted to know what exactly changed and broken this.

**Comment by hhyyg** [comment]
This behavior is probably due to this change: 

https://github.com/electron-userland/electron-builder/commit/39e678333a7f2f79e6cbd840167adee1179b64db#diff-1e1711a0cac11363d75924b56e6b1952R61-R69
https://github.com/electron-userland/electron-builder/issues/2516#issuecomment-372009092

My colleague realizes that and taught me.
